File: Backend/models/doctor.go
package models

// DoctorData is a struct to map the DoctorData model.
type DoctorData struct {
	ID             string `json:"_id" bson:"_id,omitempty"`
	Name           string `json:"Name" bson:"Name"`
	Description    string `json:"Description" bson:"Description"`
	Specialization string `json:"Specialization" bson:"Specialization"`
	ClinicAddress  string `json:"ClinicAddress" bson:"ClinicAddress"`
	ContactNumber  int    `json:"ContactNumber" bson:"ContactNumber"`
	Email          string `json:"Email" bson:"Email"`
	ImageURL       string `json:"ImageURL" bson:"ImageURL"`
	Docter_Code    string `json:"Docter_Code" bson:"Docter_Code"`
	IsVerified     bool   `json:"IsVerified" bson:"IsVerified"`
}

File: src/lib/models/Doctor.ts
import mongoose, { Document, Schema } from 'mongoose'

export interface IDoctor extends Document {
  Name: string
  Description: string
  Specialization: string
  ClinicAddress: string
  ContactNumber: number
  Email: string
  ImageURL: string
  Docter_Code: string
  IsVerified: boolean
}

const DoctorSchema = new Schema<IDoctor>(
  {
    Name: String,
    Description: String,
    Specialization: String,
    ClinicAddress: String,
    ContactNumber: Number,
    Email: String,
    ImageURL: String,
    Docter_Code: String,
    IsVerified: { type: Boolean, default: false },
  },
  {
    collection: 'DoctorDB',
    timestamps: false,
  }
)

export const Doctor =
  mongoose.models.Doctor || mongoose.model<IDoctor>('Doctor', DoctorSchema)
